alfalfa_worker/lib/run_manager.py

```diff
@@ -127,95 +135,208 @@ def checkout_run(self, run_id: str) -> Run:
-                '_id': _id,
-                'site_ref': run.id,
-                'rec': entity
-            })
-        response = self.mongo_db_recs.insert_many(array_to_insert)
-        return response
+        # Create a site obj here to keep the variable active, but it is
+        # really set as index 0 of the haystack_json
+        site = None
+
+        # create all of the recs for this site
+        for index, entity in enumerate(haystack_json):
+            id = entity['id'].replace('r:', '')
+
+            # Create default writearray objects for the site. Create this on the backend
+            # to ensure that the links are created correctly and accessible from the frontend.
+            #   Only check for records tagged with writable.
+            #   This may need to be expanded to other types in the future.
+            if entity.get('writable') == 'm:':
+                cur_status = entity.pop('curStatus', None)
+                cur_val = entity.pop('curVal', None)
+                mapping = {}
+                if cur_status is not None:
+                    mapping['curStatus'] = cur_status
+                if cur_val is not None:
+                    mapping['curVal'] = cur_val
+                if mapping:
+                    # Note that run.ref_id is currently the same as site.ref_id, but
+                    # this won't be the case in the future.
+                    self.redis.hset(f'site:{run.ref_id}:rec:{id}', mapping=mapping)
+
+            if index == 0:
+                # this is the site record, store it on the site object of the
+                # database (as well as in the recs collection, for now).
+                # TODO: convert to actual data types (which requires updating the mongo schema too)
+                # TODO: FMU's might not have this data?
+                name = f"{entity.get('dis', 'Test Case').replace('s:','')} in {entity.get('geoCity', 'Unknown City').replace('s:','')}"
+                # there might be the case where the ref_id was added to the record during
+                # a "checkin" but the rest of that is not know. So get or create the Site.
+                site = Site(ref_id=run.ref_id)
+                site.name = name
+                site.haystack_raw = haystack_json
+                site.dis = entity.get('dis')
+                site.site = entity.get('site')
+                site.area = entity.get('area')
+                site.weather_ref = entity.get('weatherRef')
+                site.tz = entity.get('tz')
+                site.geo_city = entity.get('geoCity')
+                site.geo_state = entity.get('geoState')
+                site.geo_country = entity.get('geoCountry')
+                site.geo_coord = entity.get('geoCoord')
+                site.sim_status = entity.get('simStatus')
+                site.sim_type = entity.get('simType')
+                site.save()
+
+            rec = Rec(ref_id=id, site=site).save()
+            rec_instance = RecInstance(**entity)
```

Comment:
If we made `Tag` a db model we could enumerate through entities here and add them.

Reply:
I like this idea, a lot. I just kept the sub-document approach for parity-sake.